src/homepage/howTo/mail-processing.html

```diff
@@ -0,0 +1,288 @@
+                <li><b>Matchers:</b> Are per-recipient conditions for mailet execution</li>
+                <li><b>Processors:</b> Are matcher/mailet pair execution threads</li>
+              </ul>
+
+              <p> Read <a href="/server/feature-mailetcontainer.html">this</a> for more explanations of mailet container concepts.</p>
+
+              <p>One can define the mailet container content through the <a href="/server/config-mailetcontainer.html">mailetcontailer.xml</a> file.
+              Hence, we can arrange James standard components listed <a href="/server/dev-provided-mailets.html">here</a> to achieve basic logic. But what if our goal are more
+              complex? What if we need our own processing components?</p>
+
+              <p>This page will propose a 'hands on practice' how-to using James 3.0.1. We will implement a custom mailet and a custom matcher,
+              then deploy it in a James server.</p>
+
+              <p>We need to choose our use case. We will, when a mail is older than a day, write a mail to the original sender
+              to inform him about the delay, say that we are sorry, and send him a promotion code...<pre></pre></p>
+
+
+              <header class="major">
+                <h2><b>Writing custom mailets and matchers</b></h2>
+              </header>
+
+              <p>None of the matchers and mailets available in James allows us to implement what we want. We will have to
+              write our own mailet and matcher in a separated maven project depending on James Mailet API.</p>
+
+              <p>We will write a <b>IsDelayedForMoreThan</b> mathcer with a configurable delay. If the Sent Date of incoming emails is older than specified delay, then we
+              should match (return all mail recipients). Otherwise, we just return empty.</p>
+
+              <p>To ease our Job, we can rely on the <b>org.apache.james apache-mailet-base</b> maven project, which provided us a <b>GenericMatcher</b> that we can extend.</p>
+
+              <p>Here is the dependency:</p>
+
+              <pre><code>&lt;dependency&gt;
+    &lt;groupId&gt;org.apache.james&lt;/groupId&gt;
+    &lt;artifactId&gt;apache-mailet-base&lt;/artifactId&gt;
+&lt;/dependency&gt;</code></pre>
+
+              <p>The main method of a matcher is the <b>match</b> method:</p>
+
+              <pre><code>Collection&lt;MailAddress&gt; match(Mail mail) throws MessagingException;</code></pre>
+
+              <p>For us, it becomes, with <b>maxDelay</b> being previously configured:</p>
+
+              <pre><code>    private final Clock clock;
+    private Duration maxDelay;
+
+    @Override
+    public Collection&lt;MailAddress&gt; match(Mail mail) throws MessagingException {
+        Date sentDate = mail.getMessage().getSentDate();
+        Date limit = new Date(sentDate.getTime() + maxDelay.toMillis());
+
+        if (clock.instant().isAfter(limit.toInstant())) {
```

Comment:
the two last lines could be clock.instant().isAfter(senddate.toInstant().plus(maxDelay))
I you want to keep the limit variable to decompose things, you could at least use Instant.plus() to ease the addition